### PR TITLE
add default nose location / description to fhir bundle

### LINKED ISSUE #6930: Set default location code in FHIR bundles to match Covid messages
### Background
RS has noticed that our messages to the UP are missing specimen location code even though we send it in the Covid pipeline. Turns out we have a default value for the Covid pipeline but not for the UP. We want the data we send to each pipeline to match. Victor is unsure about removing the specimen location code in our messages in case it breaks something for a receiver on the Covid pipeline.

### Action requested
Make the logic for `Specimen.collection.bodySite` in our FHIR bundles match what we do for `specimen_source_site_code` in the Covid pipeline.
https://github.com/CDCgov/prime-simplereport/blob/f0455f9578acee147c0ef3ed9a5256560f82647a/backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java#L652-L655

Resource should be in this format:
```
"bodySite" : {
    "coding" : [
      {
        "system" : "http://snomed.info/sct",
        "code" : "53342003"
      }
    ],
    "text" : "Internal nose structure (body structure)"
  }
```

### Acceptance criteria
* all FHIR bundles generated should have `Specimen.collection.bodySite` populated with a SNOMED for collection location
* the SNOMED should be the collection location of the specimen type of that result. if the specimen type has no collection location, send "53342003", Internal nose structure

### PR DESCRIPTION
**Pull request by fzhao99**
# BACKEND PULL REQUEST

## Related Issue

- Fixes #6930 

## Changes Proposed

- Adds a default specimen collection location code and description to the FHIR bundle to align with the implementation in the COVID pipeline

## Additional Information

- [A follow up spike ticket](https://github.com/CDCgov/prime-simplereport/issues/6931) will look into whether we need to tweak this to something else

## Testing

- A branch with an extra log statement is deployed to dev5. 
- Verify that extra snippet is added to the FHIR bundle by default (can use the below CSV to do so)

[dev5 Boban's Org bulk upload.csv](https://github.com/CDCgov/prime-simplereport/files/13366830/dev5.Boban.s.Org.bulk.upload.csv)

```
## Running this query in Azure...
traces
| where message has "BUNDLE WITH EXTRA VALUES"

## Should give you a bundle with ... 
"bodySite" : {
    "coding" : [
      {
        "system" : "http://snomed.info/sct",
        "code" : "53342003"
      }
    ],
    "text" : "Internal nose structure (body structure)"
  }
```

---

### CODE REVIEW

**Review by emyl3 — DISMISSED** [review]
tested on dev5 -- LGTM!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java [hunk 1]
```diff
@@ -615,24 +617,28 @@ public Specimen convertToSpecimen(ConvertToSpecimenProps props) {
...
+    var collectionCoding = collectionCodeableConcept.addCoding();
+    collectionCoding.setSystem(SNOMED_CODE_SYSTEM);
+    collectionCoding.setCode(
+        Optional.of(props)
+            .map(ConvertToSpecimenProps::getCollectionCode)
+            .orElse(DEFAULT_LOCATION_CODE));
+    collectionCodeableConcept.setText(
+        Optional.of(props)
+            .map(ConvertToSpecimenProps::getCollectionName)
+            .orElse(DEFAULT_LOCATION_NAME));
```
Comment (mehansen) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java:632]: this is not in the AC - my bad!
can we please only use the default location name if we know we're using the default location code? if a location code was provided but no location name, let's just not set the text for the bodySite
that way we're not sending an incorrect location name if for some reason we have a specimen with a location code but no location name

to break down the cases:
collectionCode and collectionName provided - use those
collectionCode provided but not collectionName - use the collectionCode and don't set the text for the name
collectionCode not provided but collectionName provided - use the default values for both
neither provided - use the default values

**Review by emyl3 — APPROVED** [review]
LGTM!

**Review by mpbrown — APPROVED** [review]
Tested on dev5 and LGTM!

**Pull request merged**